### PR TITLE
buidler-refactor

### PR DESCRIPTION
**Pull request by thegostep**
- [x] migrate tests
  - [x] convert to typescript
  - [x] add use of fixtures https://ethereum-waffle.readthedocs.io/en/latest/fixtures.html
  - [x] add use of mocks https://ethereum-waffle.readthedocs.io/en/latest/mock-contract.html
- [x] update deploy scripts
  - [x] remove manual steps
  - [x] add manager-owner pattern
  - [x] add zeppelin sdk-cli compatibility
- [ ] update CI
  - [x] update travis
  - [x] update precommit hooks: https://prettier.io/docs/en/precommit.html
  - [ ] add crytic: https://github.com/crytic/slither/wiki/Upgradeability-Checks

### CODE REVIEW

**Comment by nithinkrishna** [comment]
@thegostep can you make the test file renames register as moves rather than deletion/additions.


https://stackoverflow.com/questions/433111/how-to-make-git-mark-a-deleted-and-a-new-file-as-a-file-move

**Comment by thegostep** [comment]
> @thegostep can you make the test file renames register as moves rather than deletion/additions.

I took a crack at this - unfortunately I don't think it is possible for git to track diffs between js and ts files

**Review by nithinkrishna — APPROVED** [review]

**Comment by nithinkrishna** [comment]
Looks good. But I'd say keep it in a sperate branch (development) for now.

**Comment by thegostep** [comment]
Here is a great tutorial on process for upgrades: https://forum.openzeppelin.com/t/openzeppelin-upgrades-step-by-step-tutorial-for-hardhat/3580

File: .gitignore [hunk 1]
```diff
@@ -72,3 +72,10 @@ scDebugLog
 coverage.json
 coverage/
 coverageEnv/
+
+node_modules
```
Comment (brandoniles) [inline .gitignore:76]: I also see that package-lock.json is removed. Will we still get reproducible builds?

**Review by brandoniles — COMMENTED** [review]
Thanks Stephane, I will always support typed language!

I see that package-lock.json is removed and node_modules is added to gitignore. Will we still get reproducible builds?

AFAICT, the only changes to the solidity files are formatting. I assume this matches the new linter settings. As a general note, I'd recommend separating the code changes from the test changes when the behavior is meant to be the same. It gives more confidence that neither introduces problems on their own.

**Comment by thegostep** [comment]
> I see that package-lock.json is removed and node_modules is added to gitignore. Will we still get reproducible builds?

Yes - I switched the package manager from npm to yarn. See yarn.lock for version locking

**Comment by thegostep** [comment]
> AFAICT, the only changes to the solidity files are formatting. I assume this matches the new linter settings. As a general note, I'd recommend separating the code changes from the test changes when the behavior is meant to be the same. It gives more confidence that neither introduces problems on their own.

Fair - I can split into seperate PRs if you prefer. Just say the word ✨

**Comment by thegostep** [comment]
following @brandoniles suggestion, this PR is superseded by #176 and #177

**Pull request closed without merging**